tools/rpc.toit

```diff
@@ -117,16 +118,23 @@ monitor RpcRequestQueue_:
       last_ = request
     else:
       first_ = last_ = request
-    size_++
+    unprocessed_++
 
-    while size_ > tasks_ and tasks_ < MAX_TASKS:
+    // If there are requests that could be processed by spawning more tasks,
+    // we do that now. To avoid spending too much memory on tasks, we prefer
+    // to keep some requests unprocessed and enqueued.
+    while unprocessed_ > tasks_ and tasks_ < RpcBroker.MAX_TASKS:
       tasks_++
-      task::
+      task --background::
         // The task code runs outside the monitor, so the monitor
         // is unlocked when the requests are being processed but
         // locked when the requests are being dequeued.
         try:
-          while next := remove_first: next.process
+          while next := remove_first:
+            try:
+              next.process
+            finally:
+              unprocessed_--
```

Comment:
This probably doesn't have to be in a finally-block because `next.process` should never unwind, but being a little bit defensive feels right.